test/unittests/mongoBackend/mongoUpdateContextGeo_test.cpp

```diff
@@ -493,6 +493,313 @@ TEST(mongoUpdateContextGeoRequest, updateLocAttribute)
+    EXPECT_FALSE(a2.hasField("modDate"));
+    EXPECT_FALSE(ent.hasField("location"));
+
+    utExit();
+}
+
+/* ****************************************************************************
+*
+*  - replaceNoLocAttributeWithLocAttribute
+*/
+TEST(mongoUpdateContextGeoRequest, replaceNoLocAttributeWithLocAttribute)
+{
+    utInit();
+
+    HttpStatusCode         ms;
+    UpdateContextRequest   req;
+    UpdateContextResponse  res;
+
+    /* Prepare database */
+    prepareDatabase();
+
+    /* Forge the request (from "inside" to "outside") */
+    ContextElement* ceP = new ContextElement();
+    ceP->entityId.fill("E2", "T2", "false");
+    ContextAttribute* caP = new ContextAttribute("A2", "geo:point", "2, -4");
+    caP->typeGiven = true;
+    ceP->contextAttributeVector.push_back(caP);
+    req.contextElementVector.push_back(ceP);
+    req.updateActionType = ActionTypeReplace;
+
+    /* Invoke the function in mongoBackend library */
+    servicePathVector.clear();
+    ms = mongoUpdateContext(&req, &res, "", servicePathVector, uriParams, "", "", "", V2);
+
+    /* Check response is as expected */
+    EXPECT_EQ(SccOk, ms);
+
+    EXPECT_EQ(SccOk, res.errorCode.code);
+    EXPECT_EQ("OK", res.errorCode.reasonPhrase);
+    EXPECT_EQ("", res.errorCode.details);
+
+    ASSERT_EQ(1, res.contextElementResponseVector.size());
+    /* Context Element response # 1 */
+    EXPECT_EQ("E2", RES_CER(0).entityId.id);
+    EXPECT_EQ("T2", RES_CER(0).entityId.type);
+    EXPECT_EQ("false", RES_CER(0).entityId.isPattern);
+    ASSERT_EQ(1, RES_CER(0).contextAttributeVector.size());
+    EXPECT_EQ("A2", RES_CER_ATTR(0, 0)->name);
+    EXPECT_EQ("geo:point", RES_CER_ATTR(0, 0)->type);
+    EXPECT_EQ(0, RES_CER_ATTR(0, 0)->stringValue.size());
+    ASSERT_EQ(0, RES_CER_ATTR(0, 0)->metadataVector.size());
+    EXPECT_EQ(SccOk, RES_CER_STATUS(0).code);
+    EXPECT_EQ("OK", RES_CER_STATUS(0).reasonPhrase);
+    EXPECT_EQ("", RES_CER_STATUS(0).details);
+
+    /* Check that every involved collection at MongoDB is as expected */
+    /* Note we are using EXPECT_STREQ() for some cases, as Mongo Driver returns const char*, not string
+     * objects (see http://code.google.com/p/googletest/wiki/Primer#String_Comparison) */
+
+    DBClientBase* connection = getMongoConnection();
+
+    /* entities collection */
+    BSONObj ent, attrs;
+    std::vector<BSONElement> attrNames;
+    ASSERT_EQ(2, connection->count(ENTITIES_COLL, BSONObj()));
+
+    ent = connection->findOne(ENTITIES_COLL, BSON("_id.id" << "E1" << "_id.type" << "T1"));
+    EXPECT_STREQ("E1", C_STR_FIELD(ent.getObjectField("_id"), "id"));
+    EXPECT_STREQ("T1", C_STR_FIELD(ent.getObjectField("_id"), "type"));
+    EXPECT_FALSE(ent.hasField("creDate"));
+    EXPECT_FALSE(ent.hasField("creDate"));
+    attrs = ent.getField("attrs").embeddedObject();
+    attrNames = ent.getField("attrNames").Array();
+    ASSERT_EQ(2, attrs.nFields());
+    ASSERT_EQ(2, attrNames.size());
```

Comment:
Given the request used to replace contains only an attribute ("A2", "geo:point", "2, -4") both attrs.nField() and attrNames.size() should be 1, isn't it?

Reply:
I have just seen ("E2", "T2", "false") in the request... I'm looking to the wrong place. Sorry the noise :)

NTC